runtime/bcutil/cfreader.c

```diff
@@ -2168,44 +2168,56 @@ checkAttributes(J9CfrClassFile* classfile, J9CfrAttribute** attributes, U_32 att
 
 
 /*
-	Check the class file in @classfile.
-
-	According the the JVMS 2nd ed: (1.2)
-		"Implementations of version 1.2 of the Java 2 platform can support
-		class file formats of versions in the range 45.0 through 46.0 inclusive."
-
-	According to http://access1.sun.com/SRDs/srd_repository/tools.pdf (1.3)
-		"The Java virtual machine (JVM) now accepts class files with version
-		numbers 45.3 through 47.0, inclusive."
-
-	According to http://home.ott.oti.com/teams/bluebird/doc/cldcng-f/CLDCSpecification1.1.pdf
-		"The class file format numbers used by different JDK versions are as follows:
-		- The 45.* (usually 45.3) version number identified JDK 1.1 class files
-		- The 46.* version number identifies JDK 1.2 class files.
-		- The 47.* version number identifies JDK 1.3 class files.
-		- The 48.* version number identifies JDK 1.4 class files."
-
-	- 49.* are JDK 5.0 class files (aka JDK 1.5)
-
-	Returns -1 on error, 0 on success.
-*/
+ * Java 8 allows non-zero minor versions if the major version is < 52 (Java 8).
+ * 
+ * Java 11 and up prohibit non-zero minor versions with two exceptions
+ * 	Major version is 45, or
+ * 	Major version is the same as the runtime version and the minor version is -1 (0xFFFF)
+ * 	
+ *	To be more precise, throw an error if:
+ *	- major version < 45
+ *	- or major version > maximum allowed
+ *	- or if the runtime version is Java 8 (major version 52) and
+ *		-  major version is 52 and minor version is non-zero
+ *	- or if the runtime version is greater than Java 11 and:
+ *		- [major version > 45] and [minor version is not 0 or -1 (0xFFFF)]
+ *		- or minor version == 0xffff and
+ *			(--enable-preview is not set or major version != runtime version or major version < 55)
```

Comment:
I don't think the case "runtime version is greater than Java 11" and "minor version == 0xffff" and "major version < 55" is reflected in the code. For example, is runtime version is Java 13 and major version is 55 or earlier, an error will not occur.